include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -585,6 +585,8 @@ ERROR(objc_with_embedded,none,
       "Objective-C interoperability cannot be enabled with embedded Swift.", ())
 ERROR(no_allocations_without_embedded,none,
       "-no-allocations is only applicable with embedded Swift.", ())
+ERROR(optimize_embedded_module_without_embedded,none,
+      "-optimize-embedded-module-as is only applicable with embedded Swift.", ())
```

Comment:
should this be restricted to just embedded? couldn't we perhaps also use this leaf vs lib optimization in non-embedded cases too?

Reply:
yes, it would give benefits in regular swift, too. Assuming build systems, like swifpm, support this option